libraries/AP_Hott_Telem/AP_Hott_Telem.cpp

```diff
@@ -0,0 +1,358 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+/* 
+   Graupner Hott Telemetry library
+   Hott telemetry runs at 19200 8N1 on a non-inverted half-duplex UART
+
+   With thanks to Graupner and betaflight
+*/
+
+#include "AP_Hott_Telem.h"
+
+#if HAL_HOTT_TELEM_ENABLED
+
+#include <AP_AHRS/AP_AHRS.h>
+#include <AP_BattMonitor/AP_BattMonitor.h>
+#include <AP_Baro/AP_Baro.h>
+#include <AP_RPM/AP_RPM.h>
+#include <AP_Airspeed/AP_Airspeed.h>
+#include <AP_Stats/AP_Stats.h>
+#include <AP_GPS/AP_GPS.h>
+#include <AP_RTC/AP_RTC.h>
+#include <AP_InertialSensor/AP_InertialSensor.h>
+#include <stdio.h>
+
+#define PROT_BINARY   0x80
```

Comment:
Would be nice to use an enum class here.

Reply:
there are more possible values and we deliberately don't handle all of them. I don't think an enum class is a win here